### PR TITLE
Improve introspection and logging when tests fail

### PR DESCRIPTION
**Pull request by prashantv**

### CODE REVIEW

File: testutils/test_server.go [hunk 1]
```diff
@@ -252,7 +253,7 @@ func (ts *TestServer) waitForChannelClose(ch *tchannel.Channel) {
 	started := time.Now()
 
 	var state tchannel.ChannelState
-	for i := 0; i < 50; i++ {
+	for i := 0; i < 60; i++ {
```
Comment (akshayjshah) [inline testutils/test_server.go]: Is 60 somehow meaningful, or is it just a random number that's larger than 50?

File: testutils/test_server.go [hunk 2]
```diff
@@ -28,6 +28,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/davecgh/go-spew/spew"
```
Comment (akshayjshah) [inline testutils/test_server.go]: nit: this import belongs in the block below.

File: testutils/test_server.go (hunk 1 above)
Comment (prashantv) [inline testutils/test_server.go]: Just a number larger than 50

**Comment by prashantv** [comment]
Updated with import fix

**Pull request merged**